web3/blobs.go

```diff
@@ -219,6 +219,103 @@ func (c *Contracts) NewEIP4844Transaction(
 	return signedTx, nil
 }
 
+// NewEIP4844TransactionWithNonce is like NewEIP4844Transaction but uses a provided nonce instead of fetching from RPC
+func (c *Contracts) NewEIP4844TransactionWithNonce(
```

Comment:
This is repeated (copy&pasted) code. Please don't do that. One function must call the other.